### PR TITLE
Spec URL previews

### PR DESCRIPTION
**Pull request by turt2live**
Addresses the current state of affairs. This does not include the extra behaviour described in https://github.com/matrix-org/matrix-doc/issues/674

This also addresses https://github.com/matrix-org/matrix-doc/issues/1046

### CODE REVIEW

**Comment by richvdh** [comment]
matrixbot: ok to test

File: api/client-server/content-repo.yaml [hunk 1]
```diff
@@ -184,5 +204,63 @@ paths:
...
+        - accessToken: []
+      parameters:
+        - in: query
+          type: string
+          x-example: "https://matrix.org"
+          name: url
+          description: "The URL to get a preview of"
+          required: true
+        - in: query
+          type: long
```
Comment (richvdh) [inline api/client-server/content-repo.yaml]: s/long/number/ (http://json-schema.org/work-in-progress/WIP-jsonschema-core.html#rfc.section.4.2.1)

Likewise below (https://github.com/matrix-org/matrix-doc/pull/1064/files#diff-c734d050f9c9efffd52168da0531e288R244)

Comment (turt2live) [reply api/client-server/content-repo.yaml]: /me should really read the docs more often

Updated!

**Comment by richvdh** [comment]
\o/

**Pull request merged**